src/channel.c

```diff
@@ -1894,19 +1894,10 @@ libssh2_channel_read_ex(LIBSSH2_CHANNEL *channel, int stream_id, char *buf,
                         size_t buflen)
 {
     int rc;
-    unsigned long recv_window;
 
     if(!channel)
         return LIBSSH2_ERROR_BAD_USE;
 
-    recv_window = libssh2_channel_window_read_ex(channel, NULL, NULL);
```

Comment:
@bagder Can you cast your eye over this bit.  In your commit 03ca9020756a4e16f0294e5b35e9826ee6af2364 you said

> Removed the automatic window_size adjustments from
_libssh2_channel_read() and instead all channel readers must now make
sure to enlarge the window sizes properly themselves.

What was your original rationale for doing so, as this commit appears to reverse it?  (In fact, the window size ajustments are already back in `_libssh2_channel_read`)